File: javalinfly-core/src/main/java/com/github/unldenis/javalinfly/HelloWorldHandler.java
package com.github.unldenis.javalinfly;

import io.javalin.http.Context;

@Controller(path = "/helloworld")
public class HelloWorldHandler {

  class Person {
    String name;
    int age;

    public Person(String name, int age) {
      this.name = name;
      this.age = age;
    }
  }

  @Get
  public Response<Person, String> helloPerson(Context ctx) {
    return Response.ok(new Person("Joe", 25));
  }

}

File: javalinfly-example/src/main/kotlin/com/github/unldenis/javalinfly/example/HelloWorldHandler.kt
package com.github.unldenis.javalinfly.example

import com.github.unldenis.javalinfly.Controller
import com.github.unldenis.javalinfly.Get
import com.github.unldenis.javalinfly.Response
import io.javalin.http.Context

@Controller(path = "/helloworld")
class HelloWorldHandler {
    inner class Person(var name: String, var age: Int)

    @Get
    fun helloPerson(ctx: Context): Response<Person, String> {
        return Response.ok(Person("Joe", 25))
    }
}
